app/redux/reducers/connection.js

```diff
@@ -30,11 +31,19 @@ const handshake = function (state = {}, action) {
 
 const cpuProfile = (state = {status: 'STOPPED', data: []}, action) => {
   switch (action.type) {
-    case 'DATA':
+    case CONNECTED: {
+      return Object.assign({}, state, {data: []});
+    }
+    case 'DATA': {
       if (action.from !== 'subscriber') {
-        return state;
+        return Object.assign({}, state, {status: 'STOPPED'});
+      }
+      let data = state.data.splice(0);
+      if (state.status === 'STOPPED') {
```

Comment:
What does this condition signify? Data should be cleared only on restarting profiling. (Which is how its working now anyway)